src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -171,6 +171,7 @@ case class InferenceRule(head: InferenceRuleHead,
                          q: ConjunctiveQuery,
                          weights: FactorWeight = null,
                          valueExpr: Option[Expr] = None,
+                         ruleName: Option[String] = None,
```

Comment:
Instead of keep adding these members, can we just add a `def` that looks for `@name` from the `annotations`?